Extension/src/LanguageServer/client.ts

```diff
@@ -1566,6 +1581,11 @@ export class DefaultClient implements Client {
                     if (changedSettings["legacyCompilerArgsBehavior"]) {
                         this.configuration.handleConfigurationChange();
                     }
+
+                    if (changedSettings["caseSensitiveFileSupport"]) {
+                        util.promptForReloadWindowDueToSettingsChange();
```

Comment:
Maybe only display this on Windows for now, as changing the setting between default and enabled has no effect on other platforms.